src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -78,8 +80,13 @@ public static ContactIndex parseContactIndex(String contactIndex) throws ParseEx
      */
     public static int parseInt(String intString) throws ParseException {
         String trimmedInt = intString.trim();
-        if (trimmedInt.isEmpty() || !StringUtil.isNonZeroUnsignedInteger(trimmedInt)) {
-            throw new ParseException(MESSAGE_INVALID_INDEX);
+        if (trimmedInt.isEmpty()) {
+            throw new ParseException(MESSAGE_INVALID_INT);
+        }
+        try {
+            Integer.parseInt(trimmedInt);
```

Comment:
`Integer integer = Integer.parseInt(trimmedInt);`
would have done the same, just dont need to parse again.